src/Libraries/RevitNodes/Elements/Element.cs

```diff
@@ -779,7 +779,45 @@ public IEnumerable<Element> GetJoinedElements()
                 .ToList();
         }
 
+        /// <summary>
+        /// Unjoins elements if they are joined 
+        /// </summary>
+        /// <param name="elements">List of elements to unjoin</param>
+        /// <returns>Elements that have been unjoined</returns>
+        public static List<Element> UnjoinGeometry(List<Element> elements)
+        {
+            List<Element> modifiedElements = new List<Element>();
+            for (int i = 0; i < elements.Count; i++)
+            {
+                List<Element> joinedElements = JoinGeometryUtils.GetJoinedElements(Document, elements[i].InternalElement)
+                                                                .Select(id => Document.GetElement(id).ToDSType(true))
+                                                                .ToList();
+                if (joinedElements.Count > 0)
+                {
+                    for (int j = 0; j < joinedElements.Count; j++)
+                    {
+                        JoinGeometryUtils.UnjoinGeometry(
+                            Document,
+                            elements[i].InternalElement,
+                            joinedElements[j].InternalElement);
+
+                        // check if the unjoined element is already in the modifiedElements list 
+                        // if not we add it here
+                        if (!modifiedElements.Any(item => item.Id == joinedElements[j].Id))
+                        {
+                            modifiedElements.Add(joinedElements[j]);
+                        }
 
+                    }
+                    // add the modified element to modifiedElements if its not already there
+                    if (!modifiedElements.Any(item => item.Id == elements[i].Id))
+                    {
+                        modifiedElements.Add(elements[i]);
+                    }
+                }
```

Comment:
this would transform into checking the inner list is not empty and adding it, and if it's empty, adding a `null` to list